src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -285,10 +288,62 @@ partial class BoundPropertyAccess : IPropertyReferenceExpression
         IPropertySymbol IPropertyReferenceExpression.Property => this.PropertySymbol;
        
         IExpression IMemberReferenceExpression.Instance => this.ReceiverOpt;
-       
+
+        ISymbol IMemberReferenceExpression.Member => this.PropertySymbol;
+
         protected override OperationKind ExpressionKind => OperationKind.PropertyReferenceExpression;
     }
 
+    partial class BoundEventAccess : IEventReferenceExpression
+    {
+        IEventSymbol IEventReferenceExpression.Event => this.EventSymbol;
+
+        IExpression IMemberReferenceExpression.Instance => this.ReceiverOpt;
+
+        ISymbol IMemberReferenceExpression.Member => this.EventSymbol;
+
+        protected override OperationKind ExpressionKind => OperationKind.EventReferenceExpression;
+    }
+
+    partial class BoundEventAssignmentOperator : IEventAssignmentExpression
+    {
+
+        IEventSymbol IEventAssignmentExpression.Event => this.Event;
+
+        IExpression IEventAssignmentExpression.EventInstance => this.ReceiverOpt;
+
+        IExpression IEventAssignmentExpression.HandlerValue => this.Argument;
+
+        bool IEventAssignmentExpression.Adds => this.IsAddition;
+
+        protected override OperationKind ExpressionKind => OperationKind.EventAssignmentExpression;
+    }
+
+    partial class BoundDelegateCreationExpression : IMethodBindingExpression
+    {
+        IExpression IMemberReferenceExpression.Instance
+        {
+            get
+            {
+                BoundMethodGroup methodGroup = this.Argument as BoundMethodGroup;
+                if (methodGroup != null)
+                {
+                    return methodGroup.InstanceOpt;
+                }
+
+                return null;
+            }
+        }
+
+        bool IMethodBindingExpression.IsVirtual => this.MethodOpt != null && (this.MethodOpt.IsVirtual || this.MethodOpt.IsAbstract || this.MethodOpt.IsOverride) && !this.SuppressVirtualCalls;
```

Comment:
This code is duplicated below.  Extract out a common helper for this sort of thing.